### ISSUE TITLE
`I001` triggered unexpectedly

### ISSUE THREAD

**Issue by Gabgobie** [astral-sh/ruff#27145]
### Summary

After upgrading to Ruff 0.16.0 I got a bunch of `I001`s, some of them valid, others not so much by my account.

I'm used to "presorting in blocks". What I mean by that is the following:

```py
from .mymodule import (
	foo,
	bar,

	baz,
	qux,
)
```

should be sorted as

```py
from .mymodule import (
	bar,
	foo,

	baz,
	qux,
)
```

as the blocks indicate a semantic sorting I already did by hand.
This is currently not respected and the import is instead compressed into one alphabetically sorted block.

Additionally, it's for some reason splitting `extractor` in this one:

```__init__.py
from .extractor import (
	inspect_file as inspect_file,
	process_table as process_table,
)
```

After `ruff check --fix`:

```__init__.py
from .extractor import (
	inspect_file as inspect_file,
)
from .extractor import (
	process_table as process_table,
)
```

Best,
Gab

**Comment by ntBre** [comment]
Did you have `I001` enabled in previous versions of Ruff? We added `I001` to the new set of [default rules](https://docs.astral.sh/ruff/default-rules/#isort-i) in 0.16.0. You may just  want to disable it if you didn't have it enabled before. Or you can try some of the [lint.isort](https://docs.astral.sh/ruff/settings/#lintisort) settings to obtain the style you want.

**Comment by Gabgobie** [comment]
Thanks for the quick answer.

Going by the release notes, I didn't have it active. I went with the defaults for the most part.

As a temporary fix, I added `tool.ruff.lint.ignore = ["I001"]` to my `pyproject.toml` but I just now noticed that `ruff format` would remove that blank line. I just added it earlier when I was trying to get around `I001` (maybe it was removed by an earlier format run and I didn't notice?).

Unless I'm just way off with my expectations, I'd still say that the lint and format ran incorrectly. The import shouldn't be split up and the semantically sorted blocks should be respected. Some other tools I use - specifically tombi - do so as well when sorting.

Edit: I'll take a closer look at the isort settings as you suggested.

**Comment by ntBre** [comment]
It sounds like you want to preserve your manual import formatting, which is totally fine. In that case disabling `I001` with `ignore` like you did is the right fix.

**Comment by Gabgobie** [comment]
That's not quite the case though. I'd like to consolidate the manual and automatic formatting.

Alphabetical sorting is great but sometimes there's more to it. The formatter can give me alphabetical sorting within the sections, I can do the semantics.

Take tombi for example. In my uv workspace's `pyproject.toml`, I have local and remote dependencies declared.

```pyproject.toml
dependencies = [
	# Workspace members
	"common",
	"tools",
	"utils",
	"whathaveyou",

	# External packages
	"fastapi>=2",
	"pydantic>=2",
]
```

It alphabetically sorts the subsections while keeping them intact. This is the behavior I'd expect from Ruff when there is an empty line in the import, clearly indicating some sort of separation.

Edit: I guess the correct wording would be that it respects the separate buckets.

> In that case disabling I001 with ignore like you did is the right fix.

I guess I'll keep that for now than.
